napari/utils/colormaps/colormap_utils.py

```diff
@@ -618,7 +618,30 @@ def ensure_colormap(colormap: ValidColormapArg) -> Colormap:
     """
     with AVAILABLE_COLORMAPS_LOCK:
         if isinstance(colormap, str):
-            name = colormap
+            # is a colormap with this name (default or custom) already available?
+            cmap = AVAILABLE_COLORMAPS.get(
+                colormap,
+                AVAILABLE_COLORMAPS.get(f"custom-{colormap.lower()}", None),
+            )
+            if cmap is None:
+                cmap = _colormap_from_colors(
+                    colormap, name=f"custom-{colormap.lower()}"
+                )
+                if cmap is None:
+                    cmap = vispy_or_mpl_colormap(colormap)
+
+                for cmap_ in AVAILABLE_COLORMAPS.values():
+                    if (
+                        np.array_equal(cmap_.controls, cmap.controls)
+                        and cmap_.colors.shape == cmap.colors.shape
+                        and np.all(cmap_.colors == cmap.colors)
+                        and cmap_.interpolation == cmap.interpolation
+                    ):
+                        cmap = cmap_
+                        break
+
+            name = cmap.name
```

Comment:
here you need to not use `cmap` but some other name (that will be assigned to `cmap` at the end. The line 622 gives type `Optional[Colormap]` and mypy is not able to recognize that it is no longer true after line 631.